lib/ecco/rows_converter.rb

```diff
@@ -0,0 +1,50 @@
+module Ecco
+  module RowsConverter
+    def self.convert(java_rows, type)
+      if type == "UPDATE_ROWS"
+        convert_update_rows(java_rows)
+      else
+        convert_write_or_delete_rows(java_rows)
+      end
+    end
+
+    def self.convert_write_or_delete_rows(java_rows)
+      java_rows.to_a
+    end
+
+    def self.convert_update_rows(java_rows)
+      before_update_row = java_rows.first.get_key.to_a
+      after_update_row  = java_rows.first.get_value.to_a
+
+      Array.new(1) do
+        Tuple.new(
+          key: before_update_row,
+          value: after_update_row
+        )
+      end
+    end
```

Comment:
What `java_rows` above contains is an array containing each row that was updated. Each of those rows has the value of all columns before and after the update. Because of this we can't just call `java_rows.first`, we need to loop through every row in `java_rows` and create a `Tuple` for each of them. That is, something along the lines of:

```diff
diff --git a/lib/ecco/rows_converter.rb b/lib/ecco/rows_converter.rb
index 3b4582e..7647ff2 100644
--- a/lib/ecco/rows_converter.rb
+++ b/lib/ecco/rows_converter.rb
@@ -13,10 +13,10 @@ module Ecco
     end

     def self.convert_update_rows(java_rows)
-      before_update_row = java_rows.first.get_key.to_a
-      after_update_row  = java_rows.first.get_value.to_a
+      java_rows.map do |java_row|
+        before_update_row = java_row.get_key.to_a
+        after_update_row  = java_row.get_value.to_a

-      Array.new(1) do
         Tuple.new(
           key: before_update_row,
           value: after_update_row
```

But before attempting something like that we would need a test which updates multiple rows. All tests we have now only updates one row at a time, which is why it worked before :)

Reply:
We are taking all indexes into consideration here: e0f1331